pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -543,3 +567,20 @@ func constructFilesSecret(driver string, config map[string]interface{}) (*corev1
 	}
 	return nil, nil
 }
+
+func shouldForceRemoval(job *batchv1.Job, d data.Object) bool {
+	var createJobCondition summary.Condition
+	for _, c := range summary.GetUnstructuredConditions(d) {
+		if c.Type() == createJobConditionType {
+			createJobCondition = c
+			break
+		}
+	}
+
+	if createJobCondition.Reason() != string(errors.CreateMachineError) &&
+		d.Map("metadata", "annotations")[forceRemoveMachineAnn] != "true" {
+		return false
+	}
+
+	return (job.Spec.BackoffLimit == nil && job.Status.Failed > 0) || (job.Spec.BackoffLimit != nil && job.Status.Failed > *job.Spec.BackoffLimit)
```

Comment:
Is the goal here to look to see if the job has failed? My understanding is there should be a defined job status condition that either marks succeeded or failed. https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/job-v1/#JobStatus

Also, I may be interpreting this wrong, but `.Failed` should be the number of pods that have "failed", I'd expect that check to be `job.Status.Failed >= *job.Spec.BackoffLimit`. Additionally, people have seemingly encountered problems where a job gets marked as Failed via condition while the `Failed` is < `BackoffLimit`: https://stackoverflow.com/questions/54825671/understanding-backofflimit-in-kubernetes-job

Reply:
I will use the condition instead.